src/tools/tidy/src/error_codes_check.rs

```diff
@@ -0,0 +1,183 @@
+//! Checks that all error codes have at least one test to prevent having error
+//! codes that are silently not thrown by the compiler anymore.
+
+use std::collections::HashMap;
+use std::io::{self, BufRead, BufReader};
+use std::fs::{self, DirEntry, File};
+use std::path::Path;
+
+const WHITELIST: &[&'static str] = &[
+    "E0183",
```

Comment:
Could we add a comment to this whitelist? Should we aim to remove it eventually? Is it "we know these are actually fine but we can't detect it"?

Reply:
Some of those tests can't be tested but a lot of could and **should**. I'll add a comment in that sense.